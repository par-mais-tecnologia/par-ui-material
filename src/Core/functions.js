export const isMobile = () => window.innerWidth < 769

export const importAllImages = (r) => {
  let images = {}
  r.keys().map((item) => {
    images[item.replace('./', '')] = r(item).replace(/"/g, '')
  })
  return images
}
