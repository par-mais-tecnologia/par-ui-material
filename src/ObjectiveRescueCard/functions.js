import { SeeTheme } from '../index'

export function getObjectiveType (type) {
  switch (type) {
    case 'SECURITY_RESERVE':
      return {
        preTitle: 'Do objetivo:',
        title: 'RESERVA DE EMERGÊNCIA',
        titleProps: { color: SeeTheme.styles.colors.siren, marginLeft: '5px', marginTop: '2px' },
        sTitle: 'Saldo disponível para resgatar',
        alt: 'Reserva de emergência',
        firstQuestion: 'Quanto você deseja resgatar?',
        secondQuestion: 'Qual o motivo do resgate?',
        switchText: 'Resgatar todo o valor',
        iconSrc: 'https://s3-sa-east-1.amazonaws.com/static.parmais.com.br/images/padlock.svg',
        width: 21,
        heigth: 21
      }
    case 'TIMED_OBJECTIVE':
      return {
        preTitle: 'Do objetivo:',
        title: 'SONHOS E METAS',
        titleProps: { color: SeeTheme.styles.colors.siren, marginLeft: '5px', marginTop: '2px' },
        sTitle: 'Saldo disponível para resgatar',
        alt: 'Sonhos e Metas',
        firstQuestion: 'Quanto você deseja resgatar?',
        secondQuestion: 'Qual o motivo do resgate?',
        switchText: 'Resgatar todo o valor',
        iconSrc: 'https://s3-sa-east-1.amazonaws.com/static.parmais.com.br/images/clock.svg',
        width: 21,
        heigth: 21
      }
    case 'FINANCIAL_INDEPENDENCE':
      return {
        preTitle: 'Do objetivo:',
        title: 'VIVER DE RENDA',
        titleProps: { color: SeeTheme.styles.colors.siren, marginLeft: '5px', marginTop: '2px' },
        sTitle: 'Saldo disponível para resgatar',
        alt: 'Viver de renda',
        firstQuestion: 'Quanto você deseja resgatar?',
        secondQuestion: 'Qual o motivo do resgate?',
        switchText: 'Resgatar todo o valor',
        iconSrc: 'https://s3-sa-east-1.amazonaws.com/static.parmais.com.br/images/pig.svg',
        width: 21,
        heigth: 21
      }
    case 'IMPROVE_PROFITABILITY':
      return {
        preTitle: 'Do objetivo:',
        title: 'INVESTIR MELHOR',
        titleProps: { color: SeeTheme.styles.colors.siren, marginLeft: '5px', marginTop: '2px' },
        sTitle: 'Saldo disponível para resgatar',
        alt: 'Investir melhor',
        firstQuestion: 'Quanto você deseja resgatar?',
        secondQuestion: 'Qual o motivo do resgate?',
        switchText: 'Resgatar todo o valor',
        iconSrc: 'https://s3-sa-east-1.amazonaws.com/static.parmais.com.br/images/wallet.svg',
        width: 21,
        heigth: 21
      }
    default:
      return null
  }
}

export function verifyTitle (standardType, title) {
  if (standardType) {
    if (title) {
      return title
    } else {
      return standardType.title
    }
  }
  return title
}