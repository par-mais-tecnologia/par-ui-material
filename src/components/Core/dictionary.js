export const CAMPO_OBRIGATORIO = 'Campo Obrigatório'
export const EMAIL_INVALIDO = 'E-mail inválido'
export const EMAIL = 'E-mail'
