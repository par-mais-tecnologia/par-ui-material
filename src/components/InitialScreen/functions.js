export const convertHexToRGB = (hex, opacity) => {
  if (hex.length === 4) {
    hex = hex + hex.substring(1, 4)
  }
  hex = hex.replace('#', '')
  const r = parseInt(hex.substring(0, 2), 16)
  const g = parseInt(hex.substring(2, 4), 16)
  const b = parseInt(hex.substring(4, 6), 16)
  return `rgba(${r}, ${g}, ${b}, ${opacity})`
}
