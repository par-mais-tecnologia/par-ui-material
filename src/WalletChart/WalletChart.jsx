import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Grid, Donut } from '../index'
import { getStrategiesColors, getStrategies } from './functions'
import LegendChart from '../LegendChart'

class WalletChart extends Component {
  render () {
    const { initialInvestment, wallet, legend, showLegendPercentage, labelFirstLine, labelSecondLine } = this.props

    const strategies = getStrategies(initialInvestment, wallet)

    return (
      <Grid container direction='row' alignItems='center' justify='center'>
        <Donut
          columns={getStrategies(this.props.initialInvestment, wallet)}
          colors={getStrategiesColors()}
          labelFirstLine={labelFirstLine}
          labelSecondLine={labelSecondLine} />
        <Grid className='flex db-ns w-100 w-auto-ns justify-center justify-left-ns ml0 ml5-ns'>
          {legend && <Grid className>
            <LegendChart initialInvestment={initialInvestment} strategies={strategies} showPercentage={showLegendPercentage} />
          </Grid>}
        </Grid>
      </Grid>
    )
  }
}

WalletChart.propTypes = {
  initialInvestment: PropTypes.number,
  labelFirstLine: PropTypes.string,
  labelSecondLine: PropTypes.string,
  wallet: PropTypes.arrayOf(PropTypes.object),
  legend: PropTypes.bool
}

WalletChart.defaultProps = {
  legend: true,
  labelFirstLine: '',
  labelSecondLine: ''
}

export default WalletChart
