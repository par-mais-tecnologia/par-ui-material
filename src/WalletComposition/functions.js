import { formatDecAsPercent } from '../Core/masks'

export function getWalletComposition (arr) {
  var a = []; var b = []; var prev

  let funds = []
  arr.forEach(item => {
    funds.push(item.fundName)
  })

  let funds2 = funds.sort()
  for (var i = 0; i < funds2.length; i++) {
    if (funds2[i] !== prev) {
      a.push(funds2[i])
      b.push(1)
    } else {
      b[b.length - 1]++
    }
    prev = funds2[i]
  }

  let result = [a, b]
  let total = 0
  let percentages = []

  result[1].forEach((item, i) => {
    total += item
  })

  result[1].forEach((item, i) => {
    let itemPercent = formatDecAsPercent((item / total))
    percentages.push(itemPercent)
  })

  let objRes = []
  for (i = 0; i < result[0].length; i++) {
    objRes.push({
      fundName: a[i],
      percentage: percentages[i]
    })
  }

  return objRes
}
