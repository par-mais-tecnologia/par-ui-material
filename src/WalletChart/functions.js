import * as dictionary from '../Core/dictionary'

export function getStrategiesData (strategiesWithValues, initialInvestment) {
  let strategiesArray = []

  strategiesWithValues.map(item => {
    let object = {
      name: item[0],
      amount: item[1].toFixed(2),
      percentual: ((item[1] * 100) / initialInvestment).toFixed(1)
    }

    strategiesArray.push(object)
  })

  return strategiesArray
}

export function getStrategies (initialInvestment, wallet) {
  let strategies = inicializeStrategies()

  wallet.forEach(item => {
    strategies.forEach(strategy => {
      if (strategy[0] === item.product.strategy) {
        strategy[1] += initialInvestment * item.percentual
      }
    })
  })

  return strategies
}

function inicializeStrategies () {
  return [
    [dictionary.ACOES_STRATEGIE, 0],
    [dictionary.RENDA_POS_FIXADO_STRATEGIE, 0],
    [dictionary.RENDA_PRE_FIXADO_STRATEGIE, 0],
    [dictionary.RENDA__INFLACAO_STRATEGIE, 0],
    [dictionary.MULTIMERCADO_STRATEGIE, 0],
    [dictionary.INTERNACIONAL_STRATEGIE, 0],
    [dictionary.FUNDO_IMOBILIARIO_STRATEGIE, 0],
    [dictionary.PREVIDENCIA_STRATEGIE, 0]
  ]
}

export function getStrategieColorLegend (strategieName) {
  switch (strategieName) {
    case 'Ações': return '#D4426A'
    case 'Renda Fixa Pós-Fixado': return '#94ba1d'
    case 'Renda Fixa Pré-Fixado': return '#E6BB52'
    case 'Renda Fixa Inflação': return '#FF8000'
    case 'Multimercado': return '#007CA3'
    case 'Internacional': return '#F7DC00'
    case 'Fundo Imobiliário': return '#632B7D'
    case 'Previdência': return '#BA2B31'
  }
}

export function getStrategiesColors () {
  return {
    'Ações': '#D4426A',
    'Renda Fixa Pós-Fixado': '#94ba1d',
    'Renda Fixa Pré-Fixado': '#E6BB52',
    'Renda Fixa Inflação': '#FF8000',
    'Multimercado': '#007CA3',
    'Internacional': '#F7DC00',
    'Fundo Imobiliário': '#632B7D',
    'Previdência': '#BA2B31'
  }
}

export function formatValue (value) {
  let valueString = value.toString()
  let tmpDiv = valueString.replace(/([0-9]{2})$/g, '$1')
  let tmp = tmpDiv.replace('.', ',')
  if (tmp.length > 6) {
    tmp = tmp.replace(/([0-9]{3}),([0-9]{2}$)/g, ' $1,$2')
  }
  return tmp
}

export function formatPercent (cdi) {
  let cdiPercent = ''
  if (typeof cdi === 'string') {
    cdiPercent = cdi.replace('.', ',')
  } else {
    cdiPercent = cdi.toString().replace('.', ',')
  }

  return `${cdiPercent}`
}
