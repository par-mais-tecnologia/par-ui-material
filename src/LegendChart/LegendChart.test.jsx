import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { LegendChart } from '../index'
import { getStrategies } from '../WalletChart/functions'
import { walletExample } from '../Core/mocks'

configure({ adapter: new Adapter() })

const initialInvestment = 200000
const strategies = getStrategies(initialInvestment, walletExample)

const wrapper = shallow(
  <LegendChart initialInvestment={initialInvestment} strategies={strategies} />
)

describe('LegendChart', () => {
  it('shallow LegendChart', () => {
    expect(wrapper).toBeTruthy()
  })

  it('shallow LegendChart match Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
