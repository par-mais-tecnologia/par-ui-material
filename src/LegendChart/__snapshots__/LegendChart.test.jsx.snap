// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendChart shallow LegendChart match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LegendChart
    initialInvestment={200000}
    showPercentage={true}
    strategies={
      Array [
        Array [
          "Ações",
          0,
        ],
        Array [
          "Renda Fixa Pós-Fixado",
          162800,
        ],
        Array [
          "Renda Fixa Pré-Fixado",
          0,
        ],
        Array [
          "Renda Fixa Inflação",
          0,
        ],
        Array [
          "Multimercado",
          17000,
        ],
        Array [
          "Internacional",
          0,
        ],
        Array [
          "Fundo Imobiliário",
          20200,
        ],
        Array [
          "Previdência",
          0,
        ],
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Grid
          className="mb3 ml3"
          container={true}
          direction="column"
        >
          <Grid
            alignItems="center"
            container={true}
            direction="row"
          >
            <Grid
              className="mr2"
              style={
                Object {
                  "backgroundColor": "#94ba1d",
                  "height": 15,
                  "width": 15,
                }
              }
            />
            <Grid>
              <Typography>
                81,4
                %
              </Typography>
            </Grid>
            <Grid
              className="roboto-medium"
              container={true}
              style={
                Object {
                  "color": "#94ba1d",
                }
              }
            >
              Renda Fixa Pós-Fixado
            </Grid>
          </Grid>
        </Grid>,
        <Grid
          className="mb3 ml3"
          container={true}
          direction="column"
        >
          <Grid
            alignItems="center"
            container={true}
            direction="row"
          >
            <Grid
              className="mr2"
              style={
                Object {
                  "backgroundColor": "#007CA3",
                  "height": 15,
                  "width": 15,
                }
              }
            />
            <Grid>
              <Typography>
                8,5
                %
              </Typography>
            </Grid>
            <Grid
              className="roboto-medium"
              container={true}
              style={
                Object {
                  "color": "#007CA3",
                }
              }
            >
              Multimercado
            </Grid>
          </Grid>
        </Grid>,
        <Grid
          className="mb3 ml3"
          container={true}
          direction="column"
        >
          <Grid
            alignItems="center"
            container={true}
            direction="row"
          >
            <Grid
              className="mr2"
              style={
                Object {
                  "backgroundColor": "#632B7D",
                  "height": 15,
                  "width": 15,
                }
              }
            />
            <Grid>
              <Typography>
                10,1
                %
              </Typography>
            </Grid>
            <Grid
              className="roboto-medium"
              container={true}
              style={
                Object {
                  "color": "#632B7D",
                }
              }
            >
              Fundo Imobiliário
            </Grid>
          </Grid>
        </Grid>,
      ],
      "container": true,
      "direction": "column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Renda Fixa Pós-Fixado",
        "nodeType": "class",
        "props": Object {
          "children": <Grid
            alignItems="center"
            container={true}
            direction="row"
          >
            <Grid
              className="mr2"
              style={
                Object {
                  "backgroundColor": "#94ba1d",
                  "height": 15,
                  "width": 15,
                }
              }
            />
            <Grid>
              <Typography>
                81,4
                %
              </Typography>
            </Grid>
            <Grid
              className="roboto-medium"
              container={true}
              style={
                Object {
                  "color": "#94ba1d",
                }
              }
            >
              Renda Fixa Pós-Fixado
            </Grid>
          </Grid>,
          "className": "mb3 ml3",
          "container": true,
          "direction": "column",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "center",
            "children": Array [
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#94ba1d",
                    "height": 15,
                    "width": 15,
                  }
                }
              />,
              <Grid>
                <Typography>
                  81,4
                  %
                </Typography>
              </Grid>,
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#94ba1d",
                  }
                }
              >
                Renda Fixa Pós-Fixado
              </Grid>,
            ],
            "container": true,
            "direction": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "mr2",
                "style": Object {
                  "backgroundColor": "#94ba1d",
                  "height": 15,
                  "width": 15,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Typography>
                  81,4
                  %
                </Typography>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "81,4",
                    "%",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "81,4",
                  "%",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Renda Fixa Pós-Fixado",
                "className": "roboto-medium",
                "container": true,
                "style": Object {
                  "color": "#94ba1d",
                },
              },
              "ref": null,
              "rendered": "Renda Fixa Pós-Fixado",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Multimercado",
        "nodeType": "class",
        "props": Object {
          "children": <Grid
            alignItems="center"
            container={true}
            direction="row"
          >
            <Grid
              className="mr2"
              style={
                Object {
                  "backgroundColor": "#007CA3",
                  "height": 15,
                  "width": 15,
                }
              }
            />
            <Grid>
              <Typography>
                8,5
                %
              </Typography>
            </Grid>
            <Grid
              className="roboto-medium"
              container={true}
              style={
                Object {
                  "color": "#007CA3",
                }
              }
            >
              Multimercado
            </Grid>
          </Grid>,
          "className": "mb3 ml3",
          "container": true,
          "direction": "column",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "center",
            "children": Array [
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#007CA3",
                    "height": 15,
                    "width": 15,
                  }
                }
              />,
              <Grid>
                <Typography>
                  8,5
                  %
                </Typography>
              </Grid>,
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#007CA3",
                  }
                }
              >
                Multimercado
              </Grid>,
            ],
            "container": true,
            "direction": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "mr2",
                "style": Object {
                  "backgroundColor": "#007CA3",
                  "height": 15,
                  "width": 15,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Typography>
                  8,5
                  %
                </Typography>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "8,5",
                    "%",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "8,5",
                  "%",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Multimercado",
                "className": "roboto-medium",
                "container": true,
                "style": Object {
                  "color": "#007CA3",
                },
              },
              "ref": null,
              "rendered": "Multimercado",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Fundo Imobiliário",
        "nodeType": "class",
        "props": Object {
          "children": <Grid
            alignItems="center"
            container={true}
            direction="row"
          >
            <Grid
              className="mr2"
              style={
                Object {
                  "backgroundColor": "#632B7D",
                  "height": 15,
                  "width": 15,
                }
              }
            />
            <Grid>
              <Typography>
                10,1
                %
              </Typography>
            </Grid>
            <Grid
              className="roboto-medium"
              container={true}
              style={
                Object {
                  "color": "#632B7D",
                }
              }
            >
              Fundo Imobiliário
            </Grid>
          </Grid>,
          "className": "mb3 ml3",
          "container": true,
          "direction": "column",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "center",
            "children": Array [
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#632B7D",
                    "height": 15,
                    "width": 15,
                  }
                }
              />,
              <Grid>
                <Typography>
                  10,1
                  %
                </Typography>
              </Grid>,
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#632B7D",
                  }
                }
              >
                Fundo Imobiliário
              </Grid>,
            ],
            "container": true,
            "direction": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "mr2",
                "style": Object {
                  "backgroundColor": "#632B7D",
                  "height": 15,
                  "width": 15,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Typography>
                  10,1
                  %
                </Typography>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "10,1",
                    "%",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "10,1",
                  "%",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Fundo Imobiliário",
                "className": "roboto-medium",
                "container": true,
                "style": Object {
                  "color": "#632B7D",
                },
              },
              "ref": null,
              "rendered": "Fundo Imobiliário",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Grid
            className="mb3 ml3"
            container={true}
            direction="column"
          >
            <Grid
              alignItems="center"
              container={true}
              direction="row"
            >
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#94ba1d",
                    "height": 15,
                    "width": 15,
                  }
                }
              />
              <Grid>
                <Typography>
                  81,4
                  %
                </Typography>
              </Grid>
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#94ba1d",
                  }
                }
              >
                Renda Fixa Pós-Fixado
              </Grid>
            </Grid>
          </Grid>,
          <Grid
            className="mb3 ml3"
            container={true}
            direction="column"
          >
            <Grid
              alignItems="center"
              container={true}
              direction="row"
            >
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#007CA3",
                    "height": 15,
                    "width": 15,
                  }
                }
              />
              <Grid>
                <Typography>
                  8,5
                  %
                </Typography>
              </Grid>
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#007CA3",
                  }
                }
              >
                Multimercado
              </Grid>
            </Grid>
          </Grid>,
          <Grid
            className="mb3 ml3"
            container={true}
            direction="column"
          >
            <Grid
              alignItems="center"
              container={true}
              direction="row"
            >
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#632B7D",
                    "height": 15,
                    "width": 15,
                  }
                }
              />
              <Grid>
                <Typography>
                  10,1
                  %
                </Typography>
              </Grid>
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#632B7D",
                  }
                }
              >
                Fundo Imobiliário
              </Grid>
            </Grid>
          </Grid>,
        ],
        "container": true,
        "direction": "column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Renda Fixa Pós-Fixado",
          "nodeType": "class",
          "props": Object {
            "children": <Grid
              alignItems="center"
              container={true}
              direction="row"
            >
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#94ba1d",
                    "height": 15,
                    "width": 15,
                  }
                }
              />
              <Grid>
                <Typography>
                  81,4
                  %
                </Typography>
              </Grid>
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#94ba1d",
                  }
                }
              >
                Renda Fixa Pós-Fixado
              </Grid>
            </Grid>,
            "className": "mb3 ml3",
            "container": true,
            "direction": "column",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alignItems": "center",
              "children": Array [
                <Grid
                  className="mr2"
                  style={
                    Object {
                      "backgroundColor": "#94ba1d",
                      "height": 15,
                      "width": 15,
                    }
                  }
                />,
                <Grid>
                  <Typography>
                    81,4
                    %
                  </Typography>
                </Grid>,
                <Grid
                  className="roboto-medium"
                  container={true}
                  style={
                    Object {
                      "color": "#94ba1d",
                    }
                  }
                >
                  Renda Fixa Pós-Fixado
                </Grid>,
              ],
              "container": true,
              "direction": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "mr2",
                  "style": Object {
                    "backgroundColor": "#94ba1d",
                    "height": 15,
                    "width": 15,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Typography>
                    81,4
                    %
                  </Typography>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "81,4",
                      "%",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "81,4",
                    "%",
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Renda Fixa Pós-Fixado",
                  "className": "roboto-medium",
                  "container": true,
                  "style": Object {
                    "color": "#94ba1d",
                  },
                },
                "ref": null,
                "rendered": "Renda Fixa Pós-Fixado",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Multimercado",
          "nodeType": "class",
          "props": Object {
            "children": <Grid
              alignItems="center"
              container={true}
              direction="row"
            >
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#007CA3",
                    "height": 15,
                    "width": 15,
                  }
                }
              />
              <Grid>
                <Typography>
                  8,5
                  %
                </Typography>
              </Grid>
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#007CA3",
                  }
                }
              >
                Multimercado
              </Grid>
            </Grid>,
            "className": "mb3 ml3",
            "container": true,
            "direction": "column",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alignItems": "center",
              "children": Array [
                <Grid
                  className="mr2"
                  style={
                    Object {
                      "backgroundColor": "#007CA3",
                      "height": 15,
                      "width": 15,
                    }
                  }
                />,
                <Grid>
                  <Typography>
                    8,5
                    %
                  </Typography>
                </Grid>,
                <Grid
                  className="roboto-medium"
                  container={true}
                  style={
                    Object {
                      "color": "#007CA3",
                    }
                  }
                >
                  Multimercado
                </Grid>,
              ],
              "container": true,
              "direction": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "mr2",
                  "style": Object {
                    "backgroundColor": "#007CA3",
                    "height": 15,
                    "width": 15,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Typography>
                    8,5
                    %
                  </Typography>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "8,5",
                      "%",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "8,5",
                    "%",
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Multimercado",
                  "className": "roboto-medium",
                  "container": true,
                  "style": Object {
                    "color": "#007CA3",
                  },
                },
                "ref": null,
                "rendered": "Multimercado",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Fundo Imobiliário",
          "nodeType": "class",
          "props": Object {
            "children": <Grid
              alignItems="center"
              container={true}
              direction="row"
            >
              <Grid
                className="mr2"
                style={
                  Object {
                    "backgroundColor": "#632B7D",
                    "height": 15,
                    "width": 15,
                  }
                }
              />
              <Grid>
                <Typography>
                  10,1
                  %
                </Typography>
              </Grid>
              <Grid
                className="roboto-medium"
                container={true}
                style={
                  Object {
                    "color": "#632B7D",
                  }
                }
              >
                Fundo Imobiliário
              </Grid>
            </Grid>,
            "className": "mb3 ml3",
            "container": true,
            "direction": "column",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alignItems": "center",
              "children": Array [
                <Grid
                  className="mr2"
                  style={
                    Object {
                      "backgroundColor": "#632B7D",
                      "height": 15,
                      "width": 15,
                    }
                  }
                />,
                <Grid>
                  <Typography>
                    10,1
                    %
                  </Typography>
                </Grid>,
                <Grid
                  className="roboto-medium"
                  container={true}
                  style={
                    Object {
                      "color": "#632B7D",
                    }
                  }
                >
                  Fundo Imobiliário
                </Grid>,
              ],
              "container": true,
              "direction": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "mr2",
                  "style": Object {
                    "backgroundColor": "#632B7D",
                    "height": 15,
                    "width": 15,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Typography>
                    10,1
                    %
                  </Typography>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "10,1",
                      "%",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "10,1",
                    "%",
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Fundo Imobiliário",
                  "className": "roboto-medium",
                  "container": true,
                  "style": Object {
                    "color": "#632B7D",
                  },
                },
                "ref": null,
                "rendered": "Fundo Imobiliário",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
